src/porepy/numerics/nonlinear/line_search.py

```diff
@@ -0,0 +1,697 @@
+                f_b = y[1]
+            elif ind == num_pts:
+                a = x[ind - 1]
+                f_a = y[ind - 1]
+            else:
+                a = x[ind - 1]
+                b = x[ind]
+                f_a = y[ind - 1]
+                f_b = y[ind]
+            counter += 1
+
+        return alpha, a, b
+
+    def optimum_from_spline(
+        self,
+        f: Callable,
+        a: float,
+        b: float,
+        f_a: Optional[float] = None,
+        f_b: Optional[float] = None,
+        num_pts: Optional[int] = 5,
+    ) -> tuple[float, np.ndarray, np.ndarray]:
+        """Compute the minimum/root of the spline interpolation of the function.
+
+        Parameters:
+            f: The function to be interpolated.
+            a, b: The interval.
+            f_a: The value of the function at a. If not given, it is computed.
+            f_b: The value of the function at b. If not given, it is computed.
+            num_pts: The number of sampling points in the interval [a, b].
+
+        Returns:
+            Tuple containing:
+                The minimum of the function in the interval [a, b].
+                The x values of the spline interpolation.
+                The y values of the spline interpolation.
+
+        """
+        x = np.linspace(a, b, num_pts)
+        y_list = []
+
+        for pt in x:
+            if f_a is not None and np.isclose(pt, a):
+                f_pt = f_a
+            elif f_b is not None and np.isclose(pt, b):
+                f_pt = f_b
+            else:
+                f_pt = f(pt)
+            if np.any(np.isnan(f_pt)):
+                # If we get overflow, truncate the x vector.
+                x = x[: np.where(x == pt)[0][0]]
+                break
+            # Collect function values, scalar or vector.
+            y_list.append(f_pt)
+        if isinstance(y_list[0], np.ndarray):
+            y = np.vstack(y_list)
+        else:
+            y = np.array(y_list)
+
+        def compute_minimum_from_spline(poly, a: float, b: float) -> float:
+            """Compute the minimum of one spline and postprocess the result.
+
+            If multiple minima are found, the smallest one inside the interval [a, b] is
+            returned. If no minima are found, b is returned.
+
+            Parameters:
+                poly: The spline.
+                a, b: The interval.
+
+            Returns:
+                The minimum of the spline in the interval [a, b].
+
+            """
+            min_x = poly.roots()
+            if min_x.size == 0:
+                # If no root is found, return b.
+                return b
+            else:
+                # Find smallest root inside [a, b].
+                min_x = min_x[(min_x >= a) & (min_x <= b)]
+                if min_x.size == 0:
+                    # If no root is inside the interval, return b.
+                    return b
+                else:
+                    # In case of multiple minima, return the smallest one.
```

Comment:
```suggestion
                    # In case of multiple minima, return the smallest one, which will correspond
                    # to the most conservative step length.
```
Is this correct?